src/bot/commands/misc/convert.js

```diff
@@ -27,12 +27,50 @@ async function fetchLatestRates () {
  * @param {number} baseValue
  * @param {string} baseType Base unit
  * @param {string} targetType Target unit
- * @returns {strung} Success message
+ * @returns {string} Success message
  * @throws {Error} Generic error if conversion fails
  */
 function convertUnits (baseValue, baseType, targetType) {
 	const conversion = convert(baseValue).from(baseType).to(targetType);
-	return `${baseValue}${baseType} is equal to ${conversion.toFixed(2)}${targetType}`;
+	const fractionDigits = getFractionDigits(conversion);
+	return `${baseValue.toString()}${baseType} is equal to ${conversion.toFixed(fractionDigits)}${targetType}`;
+}
+
+/**
+ * Gets the count of zeroes after the decimal point but before a non-zero number
+ * @param {number} value
+ * @returns {number} Number of zeroes
+ */
+function getCountOfZeroesAfterDecimalPoint (value) {
+	let fractionalPart = value % 1;
+	let numberOfZeroes = -1;
+
+	while (fractionalPart < 1 && fractionalPart > 0) {
+		fractionalPart *= 10;
+		numberOfZeroes++;
+	}
+	numberOfZeroes = Math.max(0, numberOfZeroes);
+
+	return numberOfZeroes;
+}
+
+/**
+ * Gets the number of fraction digits needed to display our result with toFixed
+ * @param {number} value
+ * @returns {number} Number of fraction digits
+ */
+function getFractionDigits (value) {
+	const numberOfZeroes = getCountOfZeroesAfterDecimalPoint(value);
+
+	let fractionDigits = 2;
+	if (numberOfZeroes > 0) {
+		fractionDigits = numberOfZeroes + 2;
+	}
```

Comment:
Does this imply that whole number results will be rendered with 2 trailing zeroes no matter what (e.g. `1234.00`)? Is that desirable?

Reply:
> Does this imply that whole number results will be rendered with 2 trailing zeroes no matter what (e.g. `1234.00`)? Is that desirable?

Yeah, it would. But that was the behaviour before anyway

Reply:
I could get around this by checking if there any non-zero numbers after the decimal point and if there aren't just setting it to 0

Reply:
I didn't realize this before, but I think with this code it'll always be 2 more than necessary, e.g. something like `12.5` would render as `12.500`. That's fine with me - I was concerned that it was going to do something inconsistent with whole numbers, but I just didn't read the code well enough. I don't really care whether we render trailing zeroes or not, go with whatever you think is clearer for the output.

Unrelated - looking at it again, these `if` statements could probably just be a couple `Math` calls:

```js
return Math.max(2, Math.min(numberOfZeroes + 2, 20));
```

JS doesn't really have a good clamp function built in, but I've seen chaining `max` and `min` like this in a handful of contexts so I think it might be a bit cleaner this way. I don't have a super strong preference though; if you think doing it with `if` statements is more readable, then we can keep your version.

Reply:
`12.5` would be `12.50` yes. There would be an extra zero for numbers like this. I'll try your suggestion!

Reply:
@eritbh I used your suggestion. It works. I already pushed the changes